### PR TITLE
Remove mongodb exporter for the time being

### LINKED ISSUE #4158: Remove MongoDB Exporter until licensing is resolved or similiar functionality is implemented cleanly
### Request

Remove the mongodb exporter but keep the config blocks available. Referencing #4130 that brought this to our attention.

### Use case

Removes non-compliant code until we can resolve the percona/toolkit reference.

### PR DESCRIPTION
**Pull request by jcreixell**
#### PR Description

  - We need to remove mongodb exporter until licensing issues with one of its dependencies have been resolved.
  - Keeps the configuration and boilerplate code, making the actual exporter a no-op that always returns an error.
  - Hopefully we will be able to bring it back before the next release.

#### Which issue(s) this PR fixes

Fixes #4158 

#### PR Checklist

- [x] CHANGELOG updated

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -591,6 +581,8 @@ require (
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
 )
 
+require github.com/smartystreets/goconvey v1.7.2 // indirect
```
Comment (mattdurham) [inline go.mod:584]: What did we add? Can we roll up into the above required indirect block if this is needed?

**Review by mattdurham — APPROVED** [review]
Minor nit on go.mod but looks solid otherwise.

File: go.mod (hunk 1 above)
Comment (jcreixell) [reply go.mod:584]: let me check :eye:

File: CHANGELOG.md [hunk 2]
```diff
@@ -45,6 +45,10 @@ Main (unreleased)
 
 - Fix oauth default scope in `loki.source.azure_event_hubs`. (@akselleirv)
 
+### Breaking changes
```
[the remark this answers is not in the document]
Comment (mattdurham) [reply CHANGELOG.md:48]: IMO this logically feels like a breaking change. Something that used to work no longer does and now throws a 500.

[the remark this answers is not in the document]
Comment (rfratto) [reply CHANGELOG.md:48]: If it’s a breaking change, we can’t include it in a patch :) 

Making things a no-op should be permissible, though.

[the remark this answers is not in the document]
Comment (mattdurham) [reply CHANGELOG.md:48]: I looked back at what we did for the security CVE when disabling a versioned endpoint and it was labeled under "other changes" so there is a precedent for saying breaking changes are other changes. Seems fair enough to make it other change.

**Pull request merged**